### ISSUE TITLE
`ERROR ENAMETOOLONG: name too long` when install packages

### ISSUE THREAD

**Issue by Arttse** [pnpm/pnpm#977]
### Code to reproduce the issue:
Create `package.json` file:
```json
{
  "name": "test-pnpm",
  "version": "0.0.1",
  "private": true,
  "dependencies": {
    "react": "^16.2.0",
    "react-dom": "^16.2.0",
    "react-scripts": "1.0.17"
  }
}
```
then run `pnpm i`

### Expected behavior:
Successfuly installed

### Actual behavior:
```
Adding 988 packages to node_modules
 ERROR  ENAMETOOLONG: name too long, mkdir 
'/home/arttse/test-pnpm/node_modules/.registry.npmjs.org/eslint-config-react-app/2.0.1/babel-eslint@7.2.3+es
lint-plugin-flowtype@2.39.1+eslint-plugin-import@2.8.0+eslint-plugin-jsx-a11y@5.1.1+eslint-plugin-react@7.4.
0+eslint@4.10.0'
at 0+eslint
Resolving: total 988, reused 0, downloaded 946
```

### Additional information:

 - [pnpm-debug.log](https://gist.github.com/Arttse/4fb5cf064a76accf72e5ab5a69619658)
 - `node -v` prints: v8.9.3
 - Linux Ubuntu 16.04 (4.10.0-42-generic x86_64 GNU/Linux)

**Comment by zkochan** [comment]
I assume this happens because the directory name is very long. We can probably use a hash to name that directory when the user-friendly name is longer than 100 or so characters

**Comment by zkochan** [comment]
A fix was published in [pnpm@1.25.0](https://github.com/pnpm/pnpm/releases/tag/v1.25.0)

**Comment by Arttse** [comment]
Thank you for your prompt work! :+1:

**Comment by ianitsky** [comment]
This problem back on version [8.6.9](https://github.com/pnpm/pnpm/releases/tag/v8.6.9)

**Comment by ratoi-crysty** [comment]
Seems to be fixed in 8.6.12
